fuzz/README.md

```diff
@@ -0,0 +1,102 @@
+`VirtioQueueFunction` are initialized. The `FuzzingDescriptor`s are then transformed into regular
+`Descriptor`s. A `MockSplitQueue` object is initialized. Using it and the Vector of `Descriptor`s created,
+we create multiple `DescriptorChain`s (using the `build_multiple_desc_chains` defined in `mock.rs`) with which we can
+initialize the `GuestMemory`. Then we use the initialized `GuestMemory` to create a `Queue` object. This `Queue` object
+contains a `QueueState` object on which we call the functions generated by the fuzzer.
+
+## Fuzzing virtio-queue-ser
+
+While we initialized the queue presented in the previous section from a `MockSplitQueue`, we can also create a queue
+starting from a state. For this purpose, we use `QueueState`, more specifically `QueueStateSer` (which is a
+`QueueState` that has serialization capabilities).
+
+The fuzzing process for `QueueState` can be seen below:
+
+![Fuzzing QueueState](img/QueueState.png)
+
+Besides a Vector of `FuzzingDescriptor`s, the fuzzer initializes a `FuzzingQueueStateSer` (identical to a `QueueStateSer`
+structure) and a Vector of `VirtioQueueFunction`s.
+The `GuestMemory` is initialized similarly to the process from `virtio_queue`.
+We then create a `QueueStateSer` object from the `FuzzingQueueStateSer`, and using `QueueStateSer` we create a `QueueState`
+object. We call the functions initialized by the fuzzer on the `QueueState` object, with the help of the `GuestMemory`.
+
+## Fuzzing virtio_vsock
+
+A `VsockPacket` can be created from a `DescriptorChain` or from raw pointers, and in this fuzzer we're focusing on the former.
+The vsock device uses 2 queues: one for RX and one for TX, and the unit for operating with the
+vsock device is a `VsockPacket`.
+There are two different functions that we call on a `DescriptorChain` that can create a `VsockPacket` for the RX transmission
+(by calling the `from_rx_virtq_chain` function), or a `VsockPacket` for the TX transmission (using `from_tx_virtq_chain`).
+
+The fuzzing is done similarly to that for `virtio-queue`. The process can be seen in the picture below:
+
+![Fuzzing vsock_rx](img/Vsock%20RX%20fuzzing.png)
+
+Now the `VsockPacket` can only be created from a single descriptor chain, using either `from_rx_virtq_chain`, or
+`from_tx_virtq_chain`. Once the `VsockPacket` is created, we're
+calling all the functions from the public interface.
+
+## Running the Fuzzing Session
+
+Taking as an example fuzzing for the Virtio Queue, we are working with the following modules and
+directories:
+- `common/src/virtio_queue.rs` -> defines the structures used for fuzzing and a set of tests that
+  when ran are generating a corpus for the fuzzer.
+- `corpus/virtio_queue` -> the path of the corpus used when running the fuzzing session. This is
+  also the path where we are writing the files generated by running the test cases.
+- `fuzz_targets/virtio_queue.rs` -> the path for the Virtio Queue fuzz target.
+

```

Comment:
It might be worth adding a pre-requisites section here. I found I needed to run:

  cargo install cargo-fuzz
  
first

Reply:
I noticed that there are some other outdated things in the README so I am thinking about opening a separate PR to address them along with this comment.